.circleci/config.yml

```
@@ -20,16 +20,13 @@ version: 2
         name: "Upload"
         command: |
           [ -z "$FTL_SECRET" ] && exit 0
-          FOLDER=$([ -z "$CIRCLE_TAG" ] && echo "$CIRCLE_BRANCH" || echo "$CIRCLE_TAG")
+          DIR="${CIRCLE_TAG:-${CIRCLE_BRANCH}}"
           mv pihole-FTL "${BIN_NAME}"
           sha1sum pihole-FTL-* > ${BIN_NAME}.sha1
-          wget https://ftl.pi-hole.net:8080/FTL-client
-          chmod +x ./FTL-client
-          [[ "$CIRCLE_PR_NUMBER" == "" ]] && ./FTL-client "${FOLDER}" "${BIN_NAME}" "${FTL_SECRET}"
-          [[ "$CIRCLE_PR_NUMBER" == "" ]] && ./FTL-client "${FOLDER}" "${BIN_NAME}.sha1" "${FTL_SECRET}"
-          rm ./FTL-client
+          ssh-keyscan -H $SSH_HOST >> ~/.ssh/known_hosts
+          [[ "$CIRCLE_PR_NUMBER" == "" ]] && sftp $SSH_USER@$SSH_HOST <<<$"mkdir ${DIR}"
+          [[ "$CIRCLE_PR_NUMBER" == "" ]] && sftp $SSH_USER@$SSH_HOST <<<$"put ${BIN_NAME}* ${DIR}"
```

Can we run both commands in the same connection to avoid two handshakes, etc.?

```suggestion
          if [[ "$CIRCLE_PR_NUMBER" == "" ]]; then
            sftp -b - $SSH_USER@$SSH_HOST << "-mkdir ${DIR}
            put ${BIN_NAME}* ${DIR}
          fi
```

---

The `mkdir` command may fail if the directory already exists, and there is no good way to prevent that error. Therefore, the commands are separated so the second one will execute even if the first fails.

---

https://stackoverflow.com/questions/27420116/how-to-create-directory-if-doesnt-exists-in-sftp

---

> The mkdir command may fail if the directory already exists, and there is no good way to prevent that error.

That's why I suggested `batch` mode for SFTP. In this case, we can prepend `mksir` by `-` to continue batch execution even if the `mkdir` fails.